### PR TITLE
fix/destroy-waiter: Remove waiter from store on destroy

### PR DESCRIPTION
**Pull request by seesleestak**
The docs say the destroy action should remove the waiter from the store, but I wasn't seeing that behavior. I've changed the destroy action to be in line.

### CODE REVIEW

File: src/reducer.js [hunk 1]
```diff
@@ -37,10 +37,12 @@ const reducerMap = {
     ...state,
     [payload.name]: waiter(state[payload.name], { type: t.CLEAR, payload }),
   }),
-  [t.DESTROY]: (state, payload) => ({
-    ...state,
-    [payload.name]: waiter(state[payload.name], { type: t.RESET, payload }),
```
Comment (seesleestak) [inline src/reducer.js]: This looked to be dispatching `RESET` from the `waiter-reducer`, but there isn't a `RESET` action as best as I can tell 😄

Comment (billyxs) [reply src/reducer.js]: Right, could remove destroy from the waiter-reducer as well

Comment (seesleestak) [reply src/reducer.js]: Updated

Comment (billyxs) [reply src/reducer.js]: And if you want to drop the RESET_ALL above, it's the same as DESTROY_ALL and not documented, we can get it all cleaned up

Comment (seesleestak) [reply src/reducer.js]: Nice, didn't catch that. Updated.

**Pull request merged**